src/lib/interfaces.ts

```diff
@@ -56,26 +56,90 @@ export interface ToCItem {
 /**
  * Layout interface
  */
-export interface PageContent {
-  slug: string
-  content: string
-  frontmatter: Frontmatter
-  tocItems: Array<ToCItem>
-  lastUpdatedDate?: string
-}
-
-export interface RequiredFrontmatter {
+export interface SharedFrontmatter {
   title: string
+  description: string
   lang: Lang
+  sidebarDepth?: number
+  isOutdated?: boolean
+  template?: string
 }
 
-export interface UpgradeFrontmatter extends RequiredFrontmatter {
-  description?: string
-  sidebarDepth?: number
+export interface StaticFrontmatter extends SharedFrontmatter {
+  postMergeBannerTranslation?: string
+  hideEditButton?: boolean
+}
+
+/**
+ * TODO: Refactor markdown content that currently uses SummaryPointsNumbered
+ * to use SummaryPoints (`summaryPoints: Array<string>`) instead. Then
+ * deprecate @/lib/util/getSummaryPoints.ts
+ */
+export interface SummaryPointsNumbered {
   summaryPoint1?: string
   summaryPoint2?: string
   summaryPoint3?: string
   summaryPoint4?: string
-  image?: string
-  isOutdated?: boolean
+}
+
+interface SummaryPoints {
+  summaryPoints: Array<string>
+}
+
+interface ImageInfo {
+  image: string
+  alt: string
+}
+
+export interface UpgradeFrontmatter
+  extends SharedFrontmatter,
+    SummaryPointsNumbered,
+    Omit<ImageInfo, "alt"> {}
+
+export interface RoadmapFrontmatter extends SharedFrontmatter, ImageInfo {
+  buttons: Array<{
+    label: string
+    toId?: string
+    to?: string
+    variant?: string
+  }>
+}
+
+export interface UseCasesFrontmatter
+  extends SharedFrontmatter,
+    SummaryPointsNumbered,
+    ImageInfo {
+  emoji: string
+}
+
+export interface StakingFrontmatter
+  extends SharedFrontmatter,
+    SummaryPoints,
+    ImageInfo {
+  emoji: string
+}
+
+export interface DocsFrontmatter extends SharedFrontmatter {
+  incomplete?: boolean
+  hideEditButton?: boolean
+}
+
+export interface TutorialFrontmatter extends SharedFrontmatter {
+  tags: Array<string>
+  author: string
+  source?: string
+  sourceUrl?: string
+  skill: string
+  published: string
+  address?: string
+  postMergeBannerTranslation?: string
+  hideEditButton?: boolean
+}
+
+export interface MdPageContent {
+  slug: string
+  content: string
+  frontmatter: Frontmatter
+  tocItems: Array<ToCItem>
+  lastUpdatedDate?: string
```

Comment:
Union type `Frontmatter` used here to accept all types of frontmatter with page content; cast as specific layout frontmatter type within layout file to enforce layout-specific schema.